IPython/terminal/interactiveshell.py

```diff
@@ -271,7 +269,11 @@ def patch_stdout(**kwargs):
                             history=history,
                             completer=IPythonPTCompleter(shell=self,
                                                     patch_stdout=patch_stdout),
-                            enable_history_search=True,
+                            enable_history_search=(
```

Comment:
Thanks! Can you make this a `Bool(True, config=True)` trait rather than an environment variable? That's how we make things configurable in IPython. See `highlight_matching_brackets` above on L194 for a similar example (it has a default of False, but this should default to True). Then here it can be `enable_history_search=self.enable_history_search`.

Reply:
Perfect, thanks!